devito/arch/compiler.py

```diff
@@ -597,8 +616,9 @@ def __lookup_cmds__(self):
     'pgcc': PGICompiler,
     'pgi': PGICompiler,
     'nvc': NvidiaCompiler,
-    'nvcc': NvidiaCompiler,
+    'nvc++': NvidiaCompiler,
     'nvidia': NvidiaCompiler,
+    'nvcc': CudaCompiler,
```

Comment:
would probably calll it `cuda` , nvc/nvc++ bit too close.

Reply:
that would overload `DEVITO_LANGUAGE=cuda`

I agree the names are close. But these are the NVidia names, really.

* nvcc is the NVidia CUDA Compiler
* nvc is the NVidia C compiler for C/C++/OpenACC/OpenMP/etc (but not CUDA)

Reply:
I get you but I still find the name confusing, in particular with the redundant ones (`nvidia`, `nvc`) that make it even more confusing. 
Why is it an issue to have same name for language and arch?

Reply:
Not an issue. And I don't have strong objections. So changed it to `cuda` as per your suggestion